src/index.ts

```diff
@@ -22,20 +24,29 @@ async function run() {
     const startupOptions = core.getInput("startup-options").split(" ");
 
     const helperScript = await matlab.generateScript(workspaceDir, command);
-    const execOptions  = { env: {
-        ...process.env,
-        "MW_MATLAB_BUILDTOOL_DEFAULT_PLUGINS_FCN_OVERRIDE":"ciplugins.github.getDefaultPlugins",
-    }};
+    const execOptions = {
+        env: {
+            ...process.env,
+            "MW_MATLAB_BUILDTOOL_DEFAULT_PLUGINS_FCN_OVERRIDE": "ciplugins.github.getDefaultPlugins",
+        }
+    };
 
     await matlab.runCommand(
         helperScript,
         platform,
         architecture,
-        (cmd,args)=>exec.exec(cmd,args,execOptions),
+        (cmd, args) => exec.exec(cmd, args, execOptions),
         startupOptions
     );
+
 }
 
-run().catch((e) => {
+run().catch(e => {
     core.setFailed(e);
-});
+}).finally(() => {
+    try {
+        buildSummary.processAndDisplayBuildSummary();
+    } catch (ex) {
+        console.error('An error occurred while reading the build summary file or adding the build summary table:', ex);
+    }
+});
```

Comment:
What about doing this in a finally on the `matlab.runCommand` call instead? It seems like it doesn't make sense to try to do this unless we've actually executed MATLAB.

Reply:
Done.